core/modules/qdisp/JobDescription.cc

```diff
@@ -59,7 +61,10 @@ JobDescription::JobDescription(QueryId qId, int jobId, ResourceUnit const& resou
 }
 
 
-bool JobDescription::incrAttemptCount() {
+bool JobDescription::incrAttemptCountScrubResults() {
+    if (_attemptCount >= 0) {
+        _respHandler->scrubResults(_jobId, _attemptCount);
+    }
     ++_attemptCount;
     if (_attemptCount > MAX_JOB_ATTEMPTS) {
```

Comment:
For simplicity, I'm mergin the jobId and attempt number into a single int, so it can be one number in the table. I don't think retrying a job 100 times is reasonable. I suspect 5 or 10 is more likely. Using a value of 100 makes it a lot easier for a human to read the combined id. There should be a comment at the definition of MAX_JOB_ATTEMPTS.